EIPS/eip-7492.md

```diff
@@ -0,0 +1,333 @@
+the return of information from the receiver back up the call stack.
+
+Overloaded method - call this to specify a bytes argument.
+
+###### Parameters
+
+| Name        | Type    | Description                                                                                                                                                                          |
+| ----------- | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| receiver\_  | address | The receiving address for this token spend. Contracts must implement `ERC20SpendableReceiver` to receive spendadle tokens. For more detail see `ERC20SpendableReceiver`.             |
+| spent\_     | uint256 | The amount of token being spent. This will be transfered as part of this call and provided as an argument on the call to `onERC20SpendableReceived` on the `ERC20SpendableReceiver`. |
+| arguments\_ | bytes   | Bytes argument to send with the call. See {mock} contracts for details on encoding and decoding arguments from bytes.                                                                |
+
+#### `IERC20SpendableReceiver`
+
+Implementation of `IERC20SpendableReceiver` interface.
+
+`ERC20Spendable` allows [ERC-20s](./eip-20.md) to operate as 'spendable' items, i.e. an [ERC-20](./eip-20.md) token that
+can trigger an action on another contract at the same time as being transfered. Similar to `ERC-677`
+and the hooks in [ERC-777](./eip-777.md), but with more of an empasis on interoperability (returned values) than
+`ERC-677` and specifically scoped interaction rather than the general hooks of [ERC-777](./eip-777.md).
+
+##### CallMustBeFromSpendableToken
+
+```solidity
+error CallMustBeFromSpendableToken()
+```
+
+Error {CallMustBeFromSpendableToken}. The call to this method can only be from a designated spendable token.
+
+##### `onERC20SpendableReceived`
+
+```solidity
+function onERC20SpendableReceived(address spender_, uint256 spent_, bytes arguments_) external returns (bytes4 retval_, bytes returnArguments_)
+```
+
+`onERC20SpendableReceived` External function called by `ERC20SpendableTokens`. This
+validates that the token is valid and then calls the internal {\_handleSpend} method.
+You must overried {\_handleSpend} in your contract to perform processing you wish to occur
+on token spend.
+
+This method will pass back the valid bytes4 selector and any bytes argument passed from
+{\_handleSpend}.
+
+###### Parameters
+
+| Name        | Type    | Description                               |
+| ----------- | ------- | ----------------------------------------- |
+| spender\_   | address | The address spending the `ERC20Spendable` |
+| spent\_     | uint256 | The amount of token spent                 |
+| arguments\_ | bytes   | Bytes sent with the call                  |
```

Comment:
Why don't you just use a solidity interface (like your IERC20Spendable.sol!) annotated with NatSpec as your specification? You're overcomplicating things here. See [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155#specification) for an example of a more modern spec.